### PR TITLE
Allow selection of encryption type (LUKS version)

### PR DESCRIPTION
**Pull request by vojtechtrefny**
Blivet now supports both LUKS1 and LUKS2 so we should allow
choosing the "type" of encryption which will be used.

All "encryption" widgets from AddDialog were also moved to a new
EncryptionChooser widget.

### CODE REVIEW

File: blivetgui/dialogs/widgets.py [hunk 1]
```diff
@@ -211,3 +224,127 @@ def autoselect(self, device_type):
...
+        self._combobox_type = self._builder.get_object("combobox_type")
+        self._encrypt_check = self._builder.get_object("check_encrypt")
+        self._passphrase_entry = self._builder.get_object("entry_passphrase")
+        self._repeat_entry = self._builder.get_object("entry_repeat")
+
+        # always visible widgets -- encrypt checkbox and its label
+        self._visible_widgets = [self._builder.get_object("label_encrypt"),
+                                 self._encrypt_check]
+
+        # "advanced" widgets -- visible only when enrypt is checked
```
Comment (vpodzime) [inline blivetgui/dialogs/widgets.py]: *enrypt*

File: tests/blivetgui_tests/widgets_test.py [hunk 2]
```diff
@@ -76,5 +82,73 @@ def test_20_selection(self):
...
+
+        # passphrases specified and matches
+        encrypt_chooser.encrypt = True
+        encrypt_chooser._passphrase_entry.set_text("aaaaa")
+        encrypt_chooser._repeat_entry.set_text("aaaaa")
+        succ, msg = encrypt_chooser.validate_user_input()
+        self.assertTrue(succ)
+        self.assertIsNone(msg)
+
+        # passphrases specified but don't matche
```
Comment (vpodzime) [inline tests/blivetgui_tests/widgets_test.py]: *matche*

**Comment by vojtechtrefny** [comment]
Screenshot. Basically only change is the "Encryption type" dropdown menu.

image

**Comment by vpodzime** [comment]
LUKS good to me :smile:

**Pull request merged**